src/Widget/CacheTrait.php

```diff
@@ -0,0 +1,55 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Bolt\Widget;
+
+use Psr\SimpleCache\CacheInterface;
+
+trait CacheTrait
+{
+    /** @var CacheInterface */
+    private $cache;
+
+    /** @var string */
+    private $key;
+
+    public function setCacheInterface(CacheInterface $cache): void
+    {
+        $this->cache = $cache;
+        $this->key = $this->createKey();
+    }
+
+    public function cachedInvoke(): string
```

Comment:
can't this method just overwrite __invoke() ?

Reply:
I'm going to try something else to make it more elegant. 

Overriding it would be nice, but won't work in this case. A trait can override methods from the base (extended) class, but then there's no such thing as a `::parent`.